Disabled state patterns Doc (#2083)
hashicorp/design-system: website/docs/patterns/disabled-patterns/partials/guidelines/overview.md

> We recommend avoiding the use of disabled elements. These elements cannot be navigated to or interacted with, making them inaccessible to users who rely on assistive technology. Additionally, user research has validated our recommend approach through user feedback and direct observation.

Reviewer: ```suggestion
We recommend avoiding use of disabled elements. Disabled elements cannot be navigated to or interacted with, making them inaccessible to users who rely on assistive technology. User research and feedback has also validated this recommended approach.
```

Reviewer: I think we should explain how research has validated this approach. What was the related feedback or observations?

Reviewer 2: We kept it vague because there was no specific thing we can point at. This was feedback that came from Melina which is why we chose to phrase it that way.

Reviewer 2: Updating phrasing to what you suggested and resolving.

Instead, we recommend:
- Showing the content/element as enabled and interactive.

> - Hiding the content/element but showing information about their limited access.

Reviewer: ```suggestion
- Hiding the content/element but including information explaining their limited access.
```

> - Hiding the content/element but with no provided context.

Reviewer: ```suggestion
- Hiding the content/element entirely with no extra explanation.
```

> Disabling elements should be avoided; elements that are disabled are not presented to the user with assistive technology. No user can navigate to or interact with a disabled element, which can cause confusion and frustration. Additionally, our users have given us explicit feedback that they prefer our current approach of not presenting disabled elements in the UI.

Reviewer 2: Maybe instead of saying "current approach" to just "the approach of..."

Instead, we recommend:

> - Show the content/element as enabled and usable.

Reviewer 2: Maybe this, or similar?

```suggestion
- Show the content/element as enabled and usable, but provide a half-way point that contextualizes the inability to use this feature.
```

> - Show the content/element as enabled and usable.

Reviewer 2: Actually, I',m opting to remove that last bit entirely. Because we explain in detail throughout. This should be shorter and to the point because it's an overview.
